MCB-Nucleo/Core/Src/freertos.c

```diff
@@ -0,0 +1,454 @@
+ */
+float current;
+float velocity;
+
+/* USER CODE END Variables */
+/* Definitions for defaultTask */
+osThreadId_t defaultTaskHandle;
+const osThreadAttr_t defaultTask_attributes = {
+  .name = "defaultTask",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for GetADCValues */
+osThreadId_t GetADCValuesHandle;
+const osThreadAttr_t GetADCValues_attributes = {
+  .name = "GetADCValues",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for updateFlags */
+osThreadId_t updateFlagsHandle;
+const osThreadAttr_t updateFlags_attributes = {
+  .name = "updateFlags",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for motorStateMachi */
+osThreadId_t motorStateMachiHandle;
+const osThreadAttr_t motorStateMachi_attributes = {
+  .name = "motorStateMachi",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for getCANVelocity */
+osThreadId_t getCANVelocityHandle;
+const osThreadAttr_t getCANVelocity_attributes = {
+  .name = "getCANVelocity",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for getCANBatterySO */
+osThreadId_t getCANBatterySOHandle;
+const osThreadAttr_t getCANBatterySO_attributes = {
+  .name = "getCANBatterySO",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for updateState */
+osThreadId_t updateStateHandle;
+const osThreadAttr_t updateState_attributes = {
+  .name = "updateState",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+
+/* Private function prototypes -----------------------------------------------*/
+/* USER CODE BEGIN FunctionPrototypes */
+
+/* USER CODE END FunctionPrototypes */
+
+void StartDefaultTask(void *argument);
+void getADCValues(void *argument);
+void startUpdateFlags(void *argument);
+void startMotorStateMachine(void *argument);
+void startGetCANVelocity(void *argument);
+void StartSetCANBatterySO(void *argument);
+void StartUpdateState(void *argument);
+
+void MX_FREERTOS_Init(void); /* (MISRA C 2004 rule 8.1) */
+
+/**
+  * @brief  FreeRTOS initialization
+  * @param  None
+  * @retval None
+  */
+void MX_FREERTOS_Init(void) {
+  /* USER CODE BEGIN Init */
+  /* USER CODE END Init */
+
+  /* USER CODE BEGIN RTOS_MUTEX */
+  /* add mutexes, ... */
+  /* USER CODE END RTOS_MUTEX */
+
+  /* USER CODE BEGIN RTOS_SEMAPHORES */
+  /* add semaphores, ... */
+  /* USER CODE END RTOS_SEMAPHORES */
+
+  /* USER CODE BEGIN RTOS_TIMERS */
+  /* start timers, add new ones, ... */
+  /* USER CODE END RTOS_TIMERS */
+
+  /* USER CODE BEGIN RTOS_QUEUES */
+  /* add queues, ... */
+  /* USER CODE END RTOS_QUEUES */
+
+  /* Create the thread(s) */
+  /* creation of defaultTask */
+  defaultTaskHandle = osThreadNew(StartDefaultTask, NULL, &defaultTask_attributes);
```

Comment:
![image](https://github.com/UBC-Solar/firmware_v3/assets/87883655/741b50dd-73ec-4de4-a055-64a759889ee3)
From what I understand, the default task cannot be deleted since it handles initializations for freeRTOS. I'm using it as a heartbeat LED for an easier way to see if the code is running. I think it should be fine to deleted some task and place its code into the default task.

Reply:
Yeah correct. I meant in the code you can comment out the function to create the thread so that it doesn't run. The cube generator is just a tool but you can modify the code to do whatever you want after generation. For the final code of the car you could just delete all the default task code instead of commenting it out.